mindsdb/integrations/handlers/hubspot_handler/hubspot_handler.py

```diff
@@ -49,48 +182,718 @@ def __init__(self, name: str, **kwargs):
+                - ordinal_position: Position in the table
+                - description: Human-readable description
+                - original_name: Original property name from HubSpot
+
+        Raises:
+            Exception: If there's a permissions error accessing the table
+        """
+        try:
+            # Fetch sample data based on table type
+            sample_data = None
+            if table_name == "companies":
+                sample_data = list(self.connection.crm.companies.get_all(limit=sample_size))
+            elif table_name == "contacts":
+                sample_data = list(self.connection.crm.contacts.get_all(limit=sample_size))
+            elif table_name == "deals":
+                sample_data = list(self.connection.crm.deals.get_all(limit=sample_size))
+
+            if not sample_data or len(sample_data) == 0:
+                logger.warning(f"No data available for {table_name}, using defaults")
+                return self._get_default_discovered_columns(table_name)
+
+            logger.info(f"Analyzing {len(sample_data)} records for {table_name} column discovery")
+
+            # Discover all unique properties from the sample
+            all_properties = set()
+            for item in sample_data:
+                if hasattr(item, "properties") and item.properties:
+                    all_properties.update(item.properties.keys())
+
+            discovered_columns = []
+            ordinal_position = 1
+
+            # Add 'id' column first (primary key, always present)
+            discovered_columns.append(
+                {
+                    "column_name": "id",
+                    "data_type": "VARCHAR",
+                    "is_nullable": False,
+                    "ordinal_position": ordinal_position,
+                    "description": "Unique identifier for the record (Primary Key)",
+                    "original_name": "id",
+                }
+            )
+            ordinal_position += 1
+
+            # Add columns for each discovered property
+            for prop_name in sorted(all_properties):
+                # Map property name to column name
+                column_name = prop_name
+                if prop_name == "hs_lastmodifieddate":
+                    column_name = "lastmodifieddate"
+
+                # Collect sample values for type inference and nullability detection
+                column_values = []
+                for item in sample_data:
+                    if hasattr(item, "properties") and item.properties:
+                        column_values.append(item.properties.get(prop_name))
+                    else:
+                        column_values.append(None)
+
+                # Infer data type from samples
+                data_type = self._infer_data_type_from_samples(column_values)
+
+                # Check if column has null values
+                has_null = any(v is None for v in column_values)
```

Comment:
I think better to set 'is_nullable' = None, because we can't determine if column is nullable basing on data